addons/mail/static/src/models/notification.js

```diff
@@ -35,6 +35,9 @@ registerModel({
                     });
                 }
             }
+            if ('unpartnered_email' in data) {
```

Comment:
`convertData` will be removed eventually, but for now it is correct to (re)set the value when falsy, otherwise the value could never become empty again.

Note that considering giving `undefined` to a field does nothing (and that is the default value if the key is missing), it's actually useless to check the presence of the key with a condition at all (for primitive values like this one).
We never really cleaned it up for existing occurrences though.

Reply:
Noted, thanks for the explanation @seb-odoo